### ISSUE TITLE
[question] How to theme the singleSelect options dropdown

### ISSUE THREAD

**Issue by ak-mid** [mui/mui-x#4305]
### Order ID 💳

35878

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### The problem in depth 🔍

I try to theme the singleSelect option dropdown. My goal is to add an icon in front of all menu items. Therefore I did fork the CustomInputComponent example from the docs and replaced the Rating component with a Select component.

My problem currently is that the dropdown menu items are not selectable (handleFilterChange is not triggered at all) and that the dropdown does close right after clicking on it. The reason is probably that inputRef is not set correctly, but I'm totally stuck at that point. Any help would be greatly appreciated.

I created a codesandbox with the current state of my work:

https://codesandbox.io/s/custominputcomponent-material-demo-forked-w8l935?file=/demo.js:1454-1462

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by alexfauquette** [comment]
Hi, the problem is that you try to use the <Selecte native={false} /> due to a conflict with the filter panel click-away #1314

So your component should be 
```jsx
<Selecte native>
  <option>...</option>
  <option>...</option>
  ...
</Select>
```

You can find more details and follows progress in #4127

**Comment by ak-mid** [comment]
I can't use the native <option> element and icons at the same time or am I wrong?

In my use case I got e.g. a column called status and via ```renderCell``` I modify the cells in that column so the cells only return icons for a corresponding status. 

![Bildschirmfoto 2022-04-04 um 14 05 34](https://user-images.githubusercontent.com/60788147/161540776-62d2dbb2-54a3-4256-8c4b-afda93def61a.png)

Inside of the single select filter for that status column I only got the status text, so there is no reference between the grid cell status icon and the filter status select option.

![Bildschirmfoto 2022-04-04 um 14 06 23](https://user-images.githubusercontent.com/60788147/161541063-2ec9502b-76c8-4368-8695-62e1c072c73c.png)

I would expect that there is a callback to format the filter values like we have for cells (```renderCell```) or for headers (```renderHeader ```).

**Comment by alexfauquette** [comment]
Yes, when singlSelect will support non-native Select having a `renderValueOption` callback could be a nice to have.

**Comment by alexfauquette** [comment]
@ak-mid The PR #4361 has been merged, so the next release should allow you to use the MUI `<Select />` in the filter panel

**Comment by ak-mid** [comment]
@alexfauquette Thanks for the update. I'll check it out after a new version is released.

**Comment by m4theushw** [comment]
The latest release includes #4361. But you still can't provide custom options because the option component is hardcoded to be `MenuItem`. To close this issue, we only need to add an `renderOptionValue` property to the `singleSelect` column definition. Suggestion from @alexfauquette in https://github.com/mui/mui-x/issues/4305#issuecomment-1087647302

**Comment by alexfauquette** [comment]
Without a full reproduction, it is hard to tell
I tried to add the code snippet you provided to the previous codesandbox and it works. I di not took time to check how to fix css
https://codesandbox.io/s/custominputcomponent-material-demo-forked-3yl4dh?file=/demo.js

**Comment by ak-mid** [comment]
@alexfauquette Thanks for investing the time to update the codesandbox! 
The CSS workaround from this issue [#4125](https://github.com/mui/mui-x/issues/4125#issuecomment-1069209867) was the reason why the selected filter value didn't show up. I updated the styles and now it's working.

I also did add some CSS to fix the filter panel styles and updated the codesandbox in case somebody is also looking for a solution:
https://codesandbox.io/s/custominputcomponent-material-demo-forked-cd7itj?file=/demo.js

I still got an issue that there is no default ```None``` filter value like the other native filters have. Seems like ```valueOptions``` array doesn't include it. 

On the whole the effort to theme the filter value is IMO too complicated compared to what a user tries to accomplish.
A future easier solution would be very much appreciated.

**Comment by alexfauquette** [comment]
There is not empty option because you did not set them

The default input for singleSelect filter add an empty string before the value options
https://github.com/alexfauquette/material-ui-x/blob/quick-filter/packages/grid/x-data-grid/src/components/panel/filterPanel/GridFilterInputSingleSelect.tsx#L19:L19

This in line 24 of your codeSandbox should fix it
```diff
-return valueOptions.map((option) => (
+return ["", ...valueOptions].map((option) => (
```

https://codesandbox.io/s/custominputcomponent-material-demo-forked-r131k8?file=/demo.js

**Comment by ak-mid** [comment]
Somehow I thought the empty option is set way earlier. 
I also needed to add ```value={item.value || ''}``` to check for undefined values, because else I get  ```A component is changing an uncontrolled input to be controlled```warnings in the console.

And I needed to add ```shrink``` to my InputLabel to 100% mimic a default filter label behavior.

I updated the codesandbox one more time with the complete solution

https://codesandbox.io/s/custominputcomponent-material-demo-forked-jse81u?file=/demo.js
